barretenberg/cpp/pil/vm2/bc_hashing.pil

```diff
@@ -1,13 +1,91 @@
 // Bytecode hashing.
+include "precomputed.pil";
 include "bc_decomposition.pil";
+include "poseidon2_hash.pil";
 
 // ###########
 // Design idea
 // ###########
 //
 // We have to fetch bytecode as packed field elements from bc_decomposition.pil
 // For this, we need to introduce a selector in bc_decomposition.pil activated at
-// pc == 0, 32, 64, .... for every bytecode.
+// pc == 0, 31, 63, .... for every bytecode.
 // In bc_decomposition subtrace, we reconstruct the corresponding field element.
 // Finally, we copy through a lookup/permutation based on the selector these field elements
 // to here (bc_hashing.pil) and then proceed to hashing.
+
+//   pc_index | bytecode_id | packed_field | incremental_hash |   output_hash      | latch | 
+// -----+------------+-------------+----------------+------------------+---------------+----
+//      0     |      0      |     0xabc    |        len       | H(0xabc,len) = 0x2 |   0
```

Comment:
It would be good to add comments about why we expect many forms of attack to be safeguarded by "the hash would be wrong and to craft the correct hash would be too expensive for a crypto secure hash"

For example, we are not constraining that the bytecode_id is the same inside each block.

Reply:
i've decided it's probably cheap enough that constraining bytecode_id is worth. But will add some notes regarding the freedom of certain variables and why it isnt a problem for us